pkg/log/json.go

```diff
@@ -29,8 +29,9 @@ const (
 
 //JSONWriter writes into a JSON terminal
 type JSONWriter struct {
-	out  *logrus.Logger
-	file *logrus.Entry
+	out              *logrus.Logger
+	file             *logrus.Entry
+	hasAlreadyFailed bool
```

Comment:
This is a little bit hacky. It is weird that a logger has a logic to print or not an error based on if it has already printed another one (could be a different one) and only if the format is json. We are coupling too much this logger with the installer but it can be used outside of it.

iirc, the original issue is that an error is printed twice in the installer. The last one is in the default error stage. Another option could be to remove the line 204 and not include the default error stage if empty.

Reply:
+1 this seems dangerous.